### PR TITLE
fix(background-services): resolve PHP CLI binary via PhpExecutableFinder

### LINKED ISSUE #11932: fix(background-services): SymfonyBackgroundServiceSpawner uses PHP_BINARY which is empty under Apache mod_php
## Problem

`SymfonyBackgroundServiceSpawner` defaults its PHP binary path to the `PHP_BINARY` constant (`src/Services/Background/SymfonyBackgroundServiceSpawner.php:96`):

```php
public function __construct(
    private string $projectDir,
    ?LoggerInterface $logger = null,
    private string $phpBinary = PHP_BINARY,
) {
```

Under some Apache mod_php builds (notably the Alpine `httpd-php` image OpenEMR ships in `docker/`), `PHP_BINARY` resolves to the empty string. The spawner then constructs `new Process(['', '/path/to/bin/console', ...])` and Symfony Process throws:

```
First element must contain a non-empty program name
proc_open(Array, Array, Array, '/var/www/.../', Array, Array)
src/Services/Background/SymfonyBackgroundServiceSpawner.php(164):
   Symfony\Component\Process\Process->run()
```

This fires on every `runAllDue` tick (~once a minute when a user is logged in), and surfaces as a flood of `OpenEMR.ERROR: ExceptionHandlerListener exception` lines. No background service runs at all on affected installs.

## Root cause

`PHP_BINARY` is populated by the SAPI's `executable_location`. CLI sets it to the `php` binary; FPM usually sets it to the FPM binary; Apache mod_php depends on the build and can be empty. There's no guarantee a non-empty CLI-runnable path is available from `PHP_BINARY` alone in a web SAPI.

## Impact

- All background services (UUID generation, phimail check, Email_Service, etc.) silently fail on impacted installs. Symptoms: no UUIDs assigned to new patients, no telemetry, etc.
- Recurring CRITICAL log entries that mask other failures.

## Suggested direction

Resolve the PHP CLI binary at construction time using `Symfony\Component\Process\PhpExecutableFinder` (which checks `PHP_BINARY`, `PHP_BINDIR`, `$PATH`, etc.) and fail loudly if no executable can be located. The constructor parameter remains for tests / explicit injection.

## Repro

Any Apache mod_php SAPI where `PHP_BINARY === ''`. Triggered from `BackgroundServiceRestController::runAllDue()` (`apis/routes/_rest_routes_standard.inc.php:716`).

### PR DESCRIPTION
**Pull request by kojiromike**
## Summary

`SymfonyBackgroundServiceSpawner::__construct` defaulted its `$phpBinary` parameter to the `PHP_BINARY` constant. Under the Alpine Apache mod_php build OpenEMR ships in `docker/development-easy/` (image `openemr/openemr:flex`), `PHP_BINARY` evaluates to the empty string. The spawner then constructed `new Process(['', '/path/to/bin/console', ...])` and Symfony Process threw `First element must contain a non-empty program name` on every `runAllDue` tick. No background service ran (no UUID assignment, no Email_Service, etc.), and the OpenEMR error log filled with `OpenEMR.ERROR: ExceptionHandlerListener exception` lines that masked unrelated failures.

The fix resolves the PHP CLI binary at construction time using `Symfony\Component\Process\PhpExecutableFinder`, which falls through `PHP_BINARY` env, the constant (only when the parent runs under CLI/cli-server/phpdbg), `PHP_PATH`, `PHP_BINDIR`, and `$PATH`. The constructor throws `\RuntimeException` if no binary can be found, so the failure surfaces with an actionable message instead of an opaque Process exception. The `$phpBinary` constructor parameter is preserved (now nullable) for explicit injection from tests.

## Why CI didn't catch this

Two layered gaps. Worth widening coverage in a follow-up but not in this PR (kept minimal so the cherry-pick path is unambiguous).

1. **All PHPUnit tests run under the CLI SAPI.** `apache_*` CI configurations boot the same `openemr/openemr:flex-*` image as production, but the test runner inside the container is invoked via `phpunit` (CLI), where `PHP_BINARY` is always populated. Existing spawner tests either pass `'/bin/sh'` explicitly or rely on the CLI test runner's `PHP_BINARY`. Neither path reached the empty-binary case.
2. **No API/E2E test exercises `POST /api/background_service/$run` over the web SAPI.** That endpoint is the only path where mod_php's empty `PHP_BINARY` reaches `new Process([...])` in production. `tests/Tests/Api/` contains no `BackgroundService*` test, and `e2e` does not visit the main tab long enough to fire the parasitic AJAX call.

PHPStan can't catch this either: `PHP_BINARY` is typed `string` at static-analysis time. The empty value is a SAPI runtime detail.

A reasonable follow-up: add an API test for `POST /api/background_service/$run` that runs against the apache CI configuration, so empty-`PHP_BINARY` regressions surface during the matrix instead of after deploy.

## Test plan

- [x] `composer phpunit-isolated -- --filter SymfonyBackgroundServiceSpawner` — 26 tests, 88 assertions, all green
- [x] Reproduce in a fresh `docker/development-easy/` stack on `upstream/master`: confirmed `First element must contain a non-empty program name` flooding `/var/log/apache2/error.log` once per `goRepeaterServices` interval
- [x] Apply fix, restart `openemr` container (clears opcache), reload main tab: parasitic `POST /apis/default/api/background_service/$run` returns `200` (was `500`); no further `ExceptionHandlerListener` lines
- [x] Confirmed under apache mod_php that `(new PhpExecutableFinder())->find(false)` resolves to `/usr/bin/php` even though `PHP_BINARY` constant is `""`
- [x] Pre-commit hooks: phpcs, phpstan, rector, composer-require-checker — all pass

Closes openemr/openemr#11932

### CODE REVIEW

**Comment by kojiromike** [comment]
@aisle-analyzer

File: src/Services/Background/SymfonyBackgroundServiceSpawner.php [hunk 1]
```diff
@@ -80,22 +81,52 @@
...
+    {
+        $binary = (new PhpExecutableFinder())->find(false);
+        if ($binary === false || $binary === '') {
+            throw new \RuntimeException(
+                'Cannot locate a PHP CLI binary for background service subprocesses. '
+                . 'Set the PHP_BINARY or PHP_PATH environment variable, or place a '
+                . 'php executable on PATH.',
+            );
+        }
+        return $binary;
```
[the remark this answers is not in the document]
Comment (kojiromike) [reply src/Services/Background/SymfonyBackgroundServiceSpawner.php:129]: This is a pre-existing-compromise scenario, not a fresh attack vector. The threat requires controlling `$PATH`, `PHP_BINARY`, `PHP_PATH`, or `PHP_BINDIR` on the running PHP process; an attacker with any of those already has code execution and doesn't need a background-service round trip.

On the recommendations:

1. **Pin to `PHP_BINDIR . '/php'`** — that is exactly the path `PhpExecutableFinder` falls through to on Alpine httpd-php (the #11932 repro environment), so it's not a substantive change. Hard-pinning would also break installs where the CLI isn't co-located with `PHP_BINDIR` (Homebrew, multi-version PHP, source builds), reintroducing the brittleness this PR is removing.

2. **Required application-config knob** — adds operational burden across every deployment to harden against an attacker who already owns the host. An explicit-binary-path override is a reasonable feature; it's not a security fix.

3. **Allowlist trusted directories** — too distribution-specific to ship portably. A world-writable-parent check would be a defensible defense-in-depth, but it's separable from this fix and applies equally to every other subprocess in the codebase.

`PhpExecutableFinder` is the Symfony-recommended way to locate the PHP CLI. Using it here is appropriate; no change.
